qutebrowser/browser/hints.py

```diff
@@ -986,3 +1003,112 @@ def on_mode_left(self, mode):
             # hinting.
             return
         self._cleanup()
+
+
+class WordHinter(object):
+
+    """Generator for word hints.
+
+    Class attributes:
+
+    Attributes:
+
+    """
+
+    TAG_EXTRACTORS = {
+        "alt": lambda elem: elem["alt"],
+        "name": lambda elem: elem["name"],
+        "title": lambda elem: elem["title"],
+        "src": lambda elem: elem["src"].split('/')[-1],
+        "href": lambda elem: elem["href"].split('/')[-1],
+        "text": str,
+    }
+
+    TAGS_FOR = collections.defaultdict(
+        list, {
+            "IMG": ["alt", "title", "src"],
+            "A": ["title", "href", "text"],
+            "INPUT": ["name"]
+        }
+    )
+
+    FIRST_ALPHABETIC = re.compile('[A-Za-z]{3,}')
```

Comment:
I still think those should be moved to the places where they are used - no point in having them as class attributes.